src/plugins/posix/aio_queue.cpp

```diff
@@ -93,20 +99,19 @@ nixl_status_t aioQueue::checkCompleted() {
         return NIXL_SUCCESS;
 
     // Check all submitted I/Os
-    for (auto& aiocb : aiocbs) {
-        if (aiocb.aio_fildes == 0 || aiocb.aio_nbytes == 0)
-            continue;  // Skip unused control blocks
+    for (int i = 0; i < num_entries; i++) {
+        if (completed[i] || aiocbs[i].aio_fildes == 0 || aiocbs[i].aio_nbytes == 0)
```

Comment:
Do we still need the aio_fildes and aio_nbytes check give we don't set them anymore?

Reply:
Got it